numpy/_core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -972,11 +972,8 @@ identityhash_tester(PyObject *NPY_UNUSED(mod),
             }
         }
         else {
-            result = PyArrayIdentityHash_GetItem(tb, keys);
-            if (result == NULL) {
-                result = Py_None;
-            }
-            Py_INCREF(result);
+            // caller handles error and missing key cases
+            PyArrayIdentityHash_GetItemRef(tb, keys, &result);
```

Comment:
I guess this isn't actually used, I think I used it for a more manual test maybe?  But I think `GetItemRef` can set an error probably?  So I think you might need a `result == NULL: PyErr_Clear(); result = Py_NewRef(Py_None);`?